### PR TITLE
Fix the bug of stages in models URI being case-sensitive

### LINKED ISSUE #5311: [BUG] referencing models by stage using models URI suddenly became case sensitive
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [x] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### Describe the problem
Mlflow model registry allows users to load models using models URI. One format of the models URI is `models:/<model_name>/<stage>`, for example, `models:/regressor/Production`.

Model registry has been accepting case insensitive model URIs, like `models:/regressor/production`, in mlflow < 1.23.0.

But after the bug introduced in mlflow 1.23.0 (by PR #5027), referencing models by stage suddenly became case sensitive. To reference the latest Production version of model `regressor`, only this format is working:
* `models:/regressor/Production`

These formats are no longer working:
* `models:/regressor/production`
* `models:/regressor/pRoduction`

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [x] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by lichenran1234**
Signed-off-by: Chenran Li <chenran.li@databricks.com>

## What changes are proposed in this pull request?
Fix a bug of case-insensitive stages (e.g. `models:/regressor/pRoduction`) in models URI stopped working.

Mlflow model registry allows users to load models using models URI. One format of the models URI is `models:/<model_name>/<stage>`, for example, `models:/regressor/Production`.

Model registry has been accepting case insensitive model URIs, like `models:/regressor/production`, in mlflow < 1.23.0.

But after the bug introduced in mlflow 1.23.0 (by PR #5027), referencing models by stage suddenly became case sensitive. To reference the latest Production version of model `regressor`, only this format is working:
* `models:/regressor/Production`

These formats are no longer working:
* `models:/regressor/production`
* `models:/regressor/pRoduction`

This PR makes the stages in models URL case-insensitive again, thus fixing the bug. To make everything in sync, this PR also makes the "latest" string in `models:/<model_name>/latest` URIs case insensitive. So `Latest`, `LATEST` etc. will start to work.

## How is this patch tested?

Unit tests.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fix a bug of case-insensitive stages (e.g. `models:/regressor/pRoduction`) in models URI stopped working.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [x] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/store/artifact/utils/test_model_utils.py [hunk 1]
```diff
@@ -60,7 +62,6 @@ def test_parse_models_uri_with_latest(uri, expected_name):
         "notmodels:/NameOfModel/StageName",  # wrong scheme with stage
         "models:/",  # no model name
         "models:/Name/Stage/0",  # too many specifiers
-        "models:/Name/production",  # should be 'Production'
         "models:/Name/LATEST",  # not lower case 'latest'
```
Comment (harupy) [inline tests/store/artifact/utils/test_model_utils.py:65]: Should `"models:/Name/LATEST"` be a valid model URI or not?

Comment (lichenran1234) [reply tests/store/artifact/utils/test_model_utils.py:65]: I'm leaning towards yes. Do you think we should make "latest" case insensitive as well? I can make the change.

Comment (harupy) [reply tests/store/artifact/utils/test_model_utils.py]: > I'm leaning towards yes.

I am too :)

**Comment by harupy** [comment]
@lichenran1234 We already know what's causing the r check to fail. Please just ignore it :)

File: tests/store/artifact/utils/test_model_utils.py (hunk 1 above)
Comment (lichenran1234) [reply tests/store/artifact/utils/test_model_utils.py]: Thanks! Just added the commit

Comment (harupy) [reply tests/store/artifact/utils/test_model_utils.py]: thanks!

**Review by ankit-db — APPROVED** [review]
Thanks for the fix @lichenran1234 !

**Pull request merged**